sklearn/tests/test_kernel_ridge.py

```diff
@@ -1,85 +1,143 @@
+                        alpha=1).fit(X_reg_csr, y_reg).predict(X_reg_csr)
     assert_array_almost_equal(pred, pred2)
 
 
 def test_kernel_ridge_singular_kernel():
     # alpha=0 causes a LinAlgError in computing the dual coefficients,
     # which causes a fallback to a lstsq solver. This is tested here.
-    pred = Ridge(alpha=0, fit_intercept=False).fit(X, y).predict(X)
+    pred = Ridge(alpha=0, fit_intercept=False).fit(X_reg, y_reg).predict(X_reg)
     kr = KernelRidge(kernel="linear", alpha=0)
-    ignore_warnings(kr.fit)(X, y)
-    pred2 = kr.predict(X)
+    ignore_warnings(kr.fit)(X_reg, y_reg)
+    pred2 = kr.predict(X_reg)
     assert_array_almost_equal(pred, pred2)
 
 
-def test_kernel_ridge_precomputed():
-    for kernel in ["linear", "rbf", "poly", "cosine"]:
-        K = pairwise_kernels(X, X, metric=kernel)
-        pred = KernelRidge(kernel=kernel).fit(X, y).predict(X)
-        pred2 = KernelRidge(kernel="precomputed").fit(K, y).predict(K)
-        assert_array_almost_equal(pred, pred2)
-
-
-def test_kernel_ridge_precomputed_kernel_unchanged():
-    K = np.dot(X, X.T)
-    K2 = K.copy()
-    KernelRidge(kernel="precomputed").fit(K, y)
-    assert_array_almost_equal(K, K2)
-
-
-def test_kernel_ridge_sample_weights():
-    K = np.dot(X, X.T)  # precomputed kernel
-    sw = np.random.RandomState(0).rand(X.shape[0])
-
-    pred = Ridge(alpha=1,
-                 fit_intercept=False).fit(X, y, sample_weight=sw).predict(X)
-    pred2 = KernelRidge(kernel="linear",
-                        alpha=1).fit(X, y, sample_weight=sw).predict(X)
-    pred3 = KernelRidge(kernel="precomputed",
-                        alpha=1).fit(K, y, sample_weight=sw).predict(K)
-    assert_array_almost_equal(pred, pred2)
-    assert_array_almost_equal(pred, pred3)
-
-
 def test_kernel_ridge_multi_output():
-    pred = Ridge(alpha=1, fit_intercept=False).fit(X, Y).predict(X)
-    pred2 = KernelRidge(kernel="linear", alpha=1).fit(X, Y).predict(X)
+    pred = Ridge(alpha=1, fit_intercept=False).fit(X_reg, Y_reg).predict(X_reg)
+    pred2 = KernelRidge(kernel="linear",
+                        alpha=1).fit(X_reg, Y_reg).predict(X_reg)
     assert_array_almost_equal(pred, pred2)
 
-    pred3 = KernelRidge(kernel="linear", alpha=1).fit(X, y).predict(X)
+    pred3 = KernelRidge(kernel="linear",
+                        alpha=1).fit(X_reg, y_reg).predict(X_reg)
     pred3 = np.array([pred3, pred3]).T
     assert_array_almost_equal(pred2, pred3)
+
+
+def test_ridge_regression_classifier():
+    # This test asserts that KernelRidge and KernelRidgeClassifier
+    # have the same dual_coef_ when classification target are
+    # specified as regressions
+    krc = KernelRidgeClassifier(kernel='linear', alpha=1).fit(X, Y)
+    Y_krr = label_binarize(Y, classes=np.array([1, 2]),
+                           neg_label=-1, pos_label=1).ravel()
+    krr = KernelRidge(kernel='linear', alpha=1).fit(X, Y_krr)
+    assert_array_equal(krc.dual_coef_, krr.dual_coef_)
```

Comment:
Good sanity check. Could also please add a consistency check that `KernelRidgeClassifier(kernel='linear')` and `RidgeClassifier()` converge to the same decision function, for instance by checking that they have exactly the same hard predictions and `decision_function` ouputs on a random test set?

Could you also please add a test to check that `KernelRidgeClassifier` can reach a 100% classification accuracy on toy a non-linearly separable problem (for instance make_circle or make_swiss_roll)?